### DISCUSSION TITLE
I can call git-credential-manager-core.exe alone to communicate with github.com on Windows？

### DISCUSSION THREAD

**Discussion by xiaofuhub** [git-ecosystem/git-credential-manager#460] [Q&A] [answered]
**Feature description**

A clear and concise description of the new feature.

**Comment by mjcheetham** [comment, marked as answer]
I would first ask _why are you wanting to call GCM directly?_ Interacting with GCM like Git does is not easy.

You can call `git-credential-manager-core.exe` directly (as opposed to via Git itself with `git credential`), but you must follow the conventions set by Git when it calls GCM.

That is to say (1) call the `get` command and pass the following over standard input:
```text
protocol=<protocol><LF>
host=<host><LF>
path=<path><LF>
username=<username><LF>
<LF>
```
..where the following placeholders are (for a remote URL of `https://mona@github.com/desktop/desktop`):
- `<protocol>` is the scheme part of the remote URL (the bit before `://`), for example `https`
- `<host>` is the hostname (and optional port) from the remote URL, for example `github.com`
- `<path>` is the absolute path part from the remote URL, for example `desktop/desktop`
   - Note that the entire `path=` line should only be included if the `credential.useHttpPath` option has been set
- `<username>` is the optional username part of the remote URL, for example `mona`

and then use the credentials that GCM returns.

Next (**_important_**) if the credentials are OK, you must (2a) follow up with a call to `store`, passing the same information you got back from the `get` request in (1).

If the credential you get back from (1) is NOT ok, then (2b) call `erase` and pass the same information you got back from the `get` request in (1).

**Reply by stranik38** [reply]
-

**Reply by stranik38** [reply]
Území

**Comment by 16CentAstrology** [comment]
@ @ #303
